managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/EntryCacheManager.java

```diff
@@ -236,5 +238,9 @@ public int compareTo(EntryCache other) {
 
     }
 
+    public static Entry create(long ledgerId, long entryId, ByteBuf data) {
```

Comment:
yes, for non-persistent topics we want to create `Entry` with specific ledgerId-entryId and payload so, it can be pass to other entities (eg: dispatcher, consumer) as those entities APIs requires a `Entry` wrapper to pass payload/msgId information.